gulpfile.js

```diff
@@ -281,32 +282,31 @@ gulp.task('prefetch-tools', ['create-tools-dir'], function() {
 });
 
 function prefetch(bundle, targetFolder) {
-  let promises = new Set();
+  let promises = new Array();
   for (let key in reqs) {
     if (reqs[key].bundle === bundle) {
       let currentUrl = reqs[key].url;
-      let currentFile = path.join(targetFolder, key);
-      promises.add(new Promise((resolve,reject) => {
-        // if file is already downloaded, check its sha against the stored one
-        downloadAndReadSHA256(targetFolder, key + ".sha256", reqs[key].sha256sum, reject, (currentSHA256) => {
-          // console.log('[DEBUG] SHA256SUM for '+key+' = ' + currentSHA256);
-          isExistingSHA256Current(currentFile, currentSHA256, (dl) => {
-            dl ? resolve(true) : downloadFileAndCreateSha256(targetFolder, key, reqs[key].url, resolve, reject)
-          });
-        });
-      }));
+      let currentFile = path.join(targetFolder, reqs[key].filename);
+      promises.push(() => {
+        return new Promise((resolve,reject) => {
+          // if file is already downloaded, check its sha against the stored one
+          downloadAndReadSHA256(targetFolder, key + ".sha256", reqs[key].sha256sum, reject, (currentSHA256) => {
+            //console.log('[DEBUG] SHA256SUM for ' + key + ' = ' + currentSHA256);
+            isExistingSHA256Current(currentFile, currentSHA256, (dl) => {
+              dl ? resolve(true) : downloadFileAndCreateSha256(targetFolder, key, reqs[key].url, resolve, reject)
+            });
+          });
+        })
+      });
     }
   }
-  return Promise.all(promises).then((result) => {
-    if (bundle === 'yes') {
-      installerExe = resolveInstallerExePath('-bundle');
-    }
-  });
-}
+  console.log(promises.length);
+  return promises.reduce( function(pacc, fn) { return pacc.then(fn); }, Promise.resolve() );

```

Comment:
Thanks! I didn't notice that and was expecting PR build tests that. That is something what should be tested build check, because just running the build is obviously not enough. It should also verify that expected deliverables are produced.